### ISSUE TITLE
renumber through whole footnotes

### ISSUE THREAD

**Issue by henrich** [sphinx-doc/sphinx#4131]
### Problem and Expected results
If I divide files to chapters, footnote numbers are always start 1 for chapters. However, I don't want such behavior but numbering through whole chapters is better.

So please add  https://gist.github.com/tk0miya/83259d3cda6a9d5d1d7488f40a8d8396 to sphinx itself.

**Comment by tk0miya** [comment]
Agreed, it is good idea. but some people might prefer to current behavior, and some might prefer another behavior. So we have to improve it to better one.
